test/javascripts/remote_test.coffee

```diff
@@ -67,6 +67,53 @@ describe 'Remote', ->
       assert.equal "/foo/bar", request.url
       assert.equal "POST", request.method
 
+  describe 'callbacks', ->
+
+    it 'will call options.fail() on HTTP failures', (done) ->
+      server = sinon.fakeServer.create()
+      server.respondWith("POST", "/foo/bar",
+            [422, { "Content-Type": "text/html" },
+             '<div id="foo" refresh="foo">Error occured</div>'])
+      remote = new TurboGraft.Remote
+        httpRequestType: "POST"
+        httpUrl: "/foo/bar"
+        refreshOnError: "foo"
+        fail: done
```

Comment:
If this never failed, would this test just timeout? Might be worth adding the success case that throws a meaningful error.